[curl_internal] Disable all schemes except http

This avoids linker errors for undefined schemes.

Reasoning for not upstreaming this patch: Drake-specific build flavor.

--- lib/protocol.c
+++ lib/protocol.c
@@ -640,6 +640,12 @@
    present. */
 const struct Curl_scheme *Curl_getn_scheme(const char *scheme, size_t len)
 {
+#if 1  // Replaced by drake
+  if ((len == 4) && curl_strnequal(scheme, "http", 4)) {
+    return &Curl_scheme_http;
+  }
+  return NULL;
+#else
   typedef const struct Curl_scheme *(*letterfunc)(const char *ptr);
   static const letterfunc parse[] = {
     two_letter_scheme,
@@ -654,6 +660,7 @@
     return NULL;
 
   return parse[len - 2](scheme);
+#endif
 }
 
 const struct Curl_scheme *Curl_get_scheme(const char *scheme)
